### PR TITLE
Issue funder can remove users from issue

### PR DESCRIPTION
**Pull request by kziemianek**
##### Description
issue funder should be able to stop work

##### Checklist

- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)

##### Affected core subsystem(s)

- Email
- UI
- Backend

##### Testing
Email notification send not tested.

##### Refers/Fixes
Fixes: #463

### CODE REVIEW

**Comment by kziemianek** [comment]
Yes, I'll add them soon.

**Comment by mkosowsk** [comment]
@kziemianek that's a lot of commits! Perhaps a git rebase master is in order? @mbeacom

File: app/dashboard/views.py [hunk 1]
```diff
@@ -193,6 +194,63 @@ def remove_interest(request, bounty_id):
 
     return JsonResponse({'success': True})
 
+@require_POST
+@csrf_exempt
+def uninterested(request, bounty_id, profile_id):
```
Comment (kziemianek) [inline app/dashboard/views.py]: At the moment I have no idea how to test views, can't find any view tests in repo. Should I create new from the scratch?

File: app/marketing/management/commands/test_email.py [hunk 2]
```diff
@@ -36,3 +37,4 @@ def handle(self, *args, **options):
         # new_bounty_rejection(b, [settings.CONTACT_EMAIL])
         # new_bounty_acceptance(b, [settings.CONTACT_EMAIL])
         # bounty_expire_warning(b, [settings.CONTACT_EMAIL])
+        bounty_uninterested([settings.CONTACT_EMAIL, 'ksowocki@gmail.com'], b, i)
```
Comment (kziemianek) [inline app/marketing/management/commands/test_email.py]: Is this a correct mail test?

**Comment by kziemianek** [comment]
Sorry for the mess, accidentally did a merge.
Added two test, one for the path and another for mails. 
These are my first loc in python. Could not find any view tests, should I write one from the scratch ?

Thanks for advices!

File: app/dashboard/views.py [hunk 3]
```diff
@@ -193,6 +194,63 @@ def remove_interest(request, bounty_id):
...
+            status=401)
+
+    try:
+        bounty = Bounty.objects.get(pk=bounty_id)
+    except Bounty.DoesNotExist:
+        return JsonResponse({'errors': ['Bounty doesn\'t exist!']},
+                            status=401)
+
+
+    if request.session.get('handle') != bounty.bounty_owner_github_username:
```
Comment (owocki) [inline app/dashboard/views.py]: does this need to be a case insensitive compare? 🤔

File: app/dashboard/views.py (hunk 1 above)
Comment (owocki) [reply app/dashboard/views.py]: @mbeacom might know. i do not

File: app/marketing/management/commands/test_email.py (hunk 2 above)
Comment (owocki) [reply app/marketing/management/commands/test_email.py]: sure

**Comment by owocki** [comment]
code looks clean.. nice work :)

File: app/dashboard/views.py (hunk 1 above)
Comment (mbeacom) [reply app/dashboard/views.py]: @kziemianek You can hold off on the view tests. I'm going to be pushing up some view tests along with the django auth branch that'll make it easier for others to follow the pattern moving forward. I'll add a test for this view following that and ping you, so you can check it out.

File: app/assets/v2/js/shared.js [hunk 4]
```diff
@@ -199,20 +199,52 @@ var mutate_interest = function(bounty_pk, direction) {
...
+
+    data = sanitizeAPIResults(data['data']);
     $.each(data, function(index, value) {
       var profile = {
         local_avatar_url: value.local_avatar_url,
         handle: value.handle,
+        id: value.id,
         url: value.url
       };
+      var is_me_fullfiller = document.contxt['github_handle'] = bounty_owner_github_username;
```
Comment (kziemianek) [inline app/assets/v2/js/shared.js]: To be consistent, we should lowercase here too.

File: app/dashboard/views.py (hunk 3 above)
Comment (kziemianek) [reply app/dashboard/views.py]: Are You sure You want to spread this .lower() across the code, maybe there is a single place where we can lowercase both values once?

File: app/dashboard/views.py [hunk 5]
```diff
@@ -240,15 +300,14 @@ def interested_profiles(request, bounty_id):
...
-
     return JsonResponse({
         'paginator': {
             'num_pages': interests.paginator.num_pages,
         },
         'data': interests_data,
-        'profile_interested': profile_interested
+        'profile_interested': profile_interested,
+        'bounty_owner_github_username': bounty.bounty_owner_github_username,
+        'can_uninterest': bounty.is_funder(handle.lower())
```
Comment (kziemianek) [inline app/dashboard/views.py]: Removed this check from front-end, if this flag is set to true, then the uninterested button will be available.

File: app/dashboard/views.py [hunk 6]
```diff
@@ -193,6 +194,63 @@ def remove_interest(request, bounty_id):
...
+            status=401)
+
+    try:
+        bounty = Bounty.objects.get(pk=bounty_id)
+    except Bounty.DoesNotExist:
+        return JsonResponse({'errors': ['Bounty doesn\'t exist!']},
+                            status=401)
+
+
+    if not bounty.is_funder(request.session.get('handle').lower()):
```
Comment (kziemianek) [inline app/dashboard/views.py]: we already have this check in bounty model, so I will use it rather than defining it once again

**Comment by kziemianek** [comment]
Posted some changes, moved checks from front-end to back-end and reused existing logic. 
Still waiting for views tests.

Thanks for opportunity to contribute.
cheers!

**Comment by kziemianek** [comment]
@mbeacom requested tests, but as he mentioned I can hold off with views tests untill he write some.
If the commit is fine without them, then it's ok for me.

**Comment by kziemianek** [comment]
@mbeacom conflicts resolved :)

**Comment by owocki** [comment]
QAing now

**Comment by owocki** [comment]
i dont see where i can remove someone who started work as the funder.. maybe because of something with the migration to the new bounty details page?

http://bits.owocki.com/3o0G3M0o171Z/Screen%20Shot%202018-03-27%20at%208.28.14%20AM.png

also, i see a javascript error in my console:

```
shared.js:282 Uncaught ReferenceError: html is not defined
    at Object.success (shared.js:282)
    at i (jquery.js:2)
    at Object.fireWith [as resolveWith] (jquery.js:2)
    at A (jquery.js:4)
    at XMLHttpRequest.<anonymous> (jquery.js:4)
```

**Comment by mbeacom** [comment]
@kziemianek Can you take a look at the JS regarding `interested_list` ?  This was changed quite a bit in: https://github.com/gitcoinco/web/pull/691

**Comment by kziemianek** [comment]
I'll take a look at these problem in a while. 
Did not check after rebase.

**Comment by kziemianek** [comment]
Ok, I see, the view was redesigned. At the moment I have no idea where to put 'Stop work button' to let funder remove user from issue. Maybe You have idea ? This is how users lists looks now:

profile_list_as_is

**Comment by owocki** [comment]
the 'work started' feed item under 'all activity' is probably the best place i can think of 

http://bits.owocki.com/2n0q3Y2q0g0K/Screen%20Shot%202018-03-27%20at%203.54.37%20PM.png

remind me to send you an extra tip when this is all said and done since the merge with teh other PR kind of increased scope here

**Comment by kziemianek** [comment]
This is how it looks after changes:
remove_from_activity_list

File: app/assets/v2/js/shared.js [hunk 7]
```diff
@@ -218,41 +235,26 @@ var pull_interest_list = function(bounty_pk, callback) {
...
+          document.interested = true;
+        }
       });
     }
     if (started.length == 0) {
       started.push('<i class="fas fa-minus"></i>');
     }
     $('#started_owners_username').html(started);
-  });
-  $.getJSON('/actions/bounty/' + bounty_pk + '/interest/', function(data) {
```
Comment (kziemianek) [inline app/assets/v2/js/shared.js]: Redundant fetch

File: app/dashboard/views.py [hunk 8]
```diff
@@ -197,64 +198,63 @@ def remove_interest(request, bounty_id):
 
     return JsonResponse({'success': True})
 
-
+@require_POST
 @csrf_exempt
-@require_GET
-def interested_profiles(request, bounty_id):
```
Comment (kziemianek) [inline app/dashboard/views.py]: Removed this because it's not fetched from js anymore.

File: app/app/urls.py [hunk 9]
```diff
@@ -187,7 +187,7 @@
     # Interests
     path('actions/bounty/<int:bounty_id>/interest/new/', dashboard.views.new_interest, name='express-interest'),
     path('actions/bounty/<int:bounty_id>/interest/remove/', dashboard.views.remove_interest, name='remove-interest'),
-    path('actions/bounty/<int:bounty_id>/interest/', dashboard.views.interested_profiles, name='interested-profiles'),
```
Comment (kziemianek) [inline app/app/urls.py]: path not used anymore

**Comment by owocki** [comment]
testing now

**Comment by owocki** [comment]
just tested.. seems to work really well!  good job :) 

email screenshto: http://bits.owocki.com/0R3j2b3R3v3g/Screen%20Shot%202018-03-28%20at%204.53.59%20PM.png

**Comment by owocki** [comment]
@mbeacom this is ready to merge from my standpoint.  you?

**Comment by mbeacom** [comment]
Yeah, this lgtm. :shipit:

**Pull request merged**